Feat: add "parent_project_path" or "hierarchy_by_name" to data "env0_…

TomerHeber opened the pull request:
…project"

### Issue & Steps to Reproduce / Feature Request
resolves #948 

### Solution
1. Updated the project schema and added the new value.
2. Add required code to search based on path.
3. Added acceptance tests.
4. Added harness tests.

ItamarMalka commented on env0/data_project.go:
```diff
@@ -30,9 +31,16 @@ func dataProject() *schema.Resource {
+				Type:          schema.TypeString,
+				Description:   "a path of ancestors projects divided by the prefix '|'. Can be used as a filter when there are multiple subprojects with the same name under different parent projects. For example: 'App|Dev|us-east-1' will search for a project with the hierarchy 'App -> Dev -> us-east-1' ('us-east-1' being the parent)",
+				Optional:      true,
+				ConflictsWith: []string{"parent_project_name"},
```
does it not also conflict with the parent_project_id?

TomerHeber replied:
you're correct...
I think I did not do that because the parent_project_id is also a computed field, and I was worried it will cause issues.
But looks like the SDK knows how to handle that use-case, so I'll push the change. (Sometimes the Terraform SDK is buggy).

ItamarMalka commented on env0/data_project.go:
```diff
@@ -162,14 +182,64 @@ func getProjectByName(name string, parentId string, parentName string, meta inte
+		// right most element is the project itself, remove it.
+		parentIds = parentIds[:len(parentIds)-1]
+
+		matches, err := pathMatches(path, parentIds, meta)
```
mmmm that feels a bit complicated, couldn't we have filtered the projects by their hierarchy seeing if `hierarchy.contains(searchedPath)` ?
Or am I missing something?

ItamarMalka replied:
oh oh nvm I am wrong since those are IDs not names so it won't work that way 👌

ItamarMalka commented on env0/data_project.go:
```diff
@@ -162,14 +182,64 @@ func getProjectByName(name string, parentId string, parentName string, meta inte
 }
 
+func pathMatches(path, parentIds []string, meta interface{}) (bool, error) {
+	if len(path) != len(parentIds) {
```
so if we have a1|a2|a3|searchedProject
and we put a1|a2 as the path we would not find it, is that intended?

TomerHeber replied:
I tend to agree. We could support prefixes as well.
I don't know if that's what @away168 had in mind... but why not...

modifying to:
```
	if len(path) > len(parentIds) {
		return false, nil
	}
```

ItamarMalka reviewed (commented):
Looks good, just ping me when you answered the questions I left 👍

TomerHeber commented:
> Looks good, just ping me when you answered the questions I left 👍

@ItamarMalka thank you.
I have updated the PR. For the prefix match, I've added a couple of new tests.
All changes are in 40dcdef3a92ce46374dc5864a16145f96342b4d8

ItamarMalka reviewed (approved):
LGTM 🚀 👍 
@away168 let us know if there is anything else that you think is needed here 🙏
